src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs

```diff
@@ -22,6 +22,7 @@ public WebHostOptions(IConfiguration configuration)
             }
 
             Application = configuration[WebHostDefaults.ApplicationKey];
+            EnvironmentApplication = configuration[WebHostDefaults.EnvironmentApplicationKey];
```

Comment:
If that is the only use case then we don't need to have the two settings and it is sufficient to just use the `Application` setting to set the `ApplicationName` of the `IApplicationEnvironment` service. Note that you still need to adjust the `UseStartup<Type>()` to `UseStartup(typeof(Type).Assembly.GetName().Name)` in the test fixture.